client/src/components/ChangePasswordWindow/ChangePasswordWindow.component.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Form, Row, Col, Button } from "react-bootstrap";
+import { toast } from "react-toastify";
+import { logout, changePassword } from "../../redux/actions/authAction";
+import { Fragment } from "react";
+
+const ChangePasswordWindow = () => {
+  const [userCredential, setUserCredential] = useState({
+    currentPassword: "",
+    newPassword: "",
+    confirmPassword: "",
+  });
+
+  const {
+    userInfo: { token, userId },
+  } = useSelector((state) => state.auth);
+  const { errorMsg, success } = useSelector((state) => state.passwordChange);
+
+  const dispatch = useDispatch();
+
+  const handleError = () => {
+    toast("Unable to change password", { type: "error" });
+    return;
+  };
+
+  const handleSuccess = () => {
+    toast("Password Changed Successfully", { type: "success" });
+    return;
+  };
+
+  const handleChangePassword = (e) => {
+    e.preventDefault();
+    const { currentPassword, newPassword, confirmPassword } = userCredential;
+
+    if (!currentPassword || !newPassword || !confirmPassword) {
+      toast("Required Field cannot be empty", {
+        type: "info",
+      });
+      return;
+    }
+
+    if (newPassword !== confirmPassword) {
+      toast("Confirm Password do not match", {
+        type: "error",
+      });
+      return;
+    }
+
+    const userData = {
+      userId,
+      oldPassword: currentPassword,
+      newPassword,
+    };
+
+    dispatch(changePassword(token, userData));
+
+    setUserCredential({
+      currentPassword: "",
+      newPassword: "",
+      confirmPassword: "",
+    });
+
+    setTimeout(() => {
+      return dispatch(logout());
+    }, [2000]);
```

Comment:
@ranjit977 any specific reason why it's 2000?

Reply:
To show a successful message. Because it instantly pushes history to "/login" making users unaware of whether their password change action was successful or not.

Reply:
Please mark it as resolved.